container/src/main/java/org/jboss/forge/furnace/impl/modules/AddonModuleLoader.java

```diff
@@ -163,14 +163,18 @@ private ModuleSpec findAddonModule(ModuleIdentifier id)
                {
                   Builder builder = ModuleSpec.build(id);
 
-                  builder.addDependency(DependencySpec.createModuleDependencySpec(SystemClasspathSpec.ID));
-                  builder.addDependency(DependencySpec.createModuleDependencySpec(XPathJDKClasspathSpec.ID));
-                  builder.addDependency(DependencySpec.createModuleDependencySpec(JAXBJDKClasspathSpec.ID));
-                  builder.addDependency(DependencySpec.createModuleDependencySpec(CORBAClasspathSpec.ID));
-                  builder.addDependency(DependencySpec.createModuleDependencySpec(SunJDKClasspathSpec.ID));
-                  builder.addDependency(DependencySpec.createModuleDependencySpec(NashornJDKClasspathSpec.ID));
-                  builder.addDependency(DependencySpec.createModuleDependencySpec(JavaFXClasspathSpec.ID));
-                  builder.addDependency(DependencySpec.createModuleDependencySpec(AppleScriptClasspathSpec.ID));
+                  for (ModuleSpecProvider moduleSpec : getModuleProviders())
+                  {
+                      builder.addDependency(DependencySpec.createModuleDependencySpec(moduleSpec.getId()));
+                  }
+                  //builder.addDependency(DependencySpec.createModuleDependencySpec(SystemClasspathSpec.ID));
+                  //builder.addDependency(DependencySpec.createModuleDependencySpec(XPathJDKClasspathSpec.ID));
```

Comment:
There are some modulespecproviders being added later in this code that could be removed also?